### PR TITLE
Clean up pipecode and split huge file into folder structure

### PR DESCRIPTION
**Pull request by tigercat2000**
This commit cleans up any remaining relative pathing in the ATMOSPHERICS/
folder, and splits the absolutely huge pipes.dm file into 6-7 files in the
pipes/ folder tree.

@Krausus @Fox-McCloud Going to poke you two and ask for a review; I pretty much just removed every relative path and split pipes.dm, want to make sure

### CODE REVIEW

File: code/ATMOSPHERICS/components/binary_devices/binary_atmos_base.dm [hunk 1]
```diff
@@ -12,131 +12,131 @@ obj/machinery/atmospherics/binary
...
-
-		return results
-
-	disconnect(obj/machinery/atmospherics/reference)
-		if(reference==node1)
-			del(network1)
-			node1 = null
+/obj/machinery/atmospherics/binary/disconnect(obj/machinery/atmospherics/reference)
+	if(reference==node1)
+		del(network1)
```
Comment (Fox-McCloud) [inline code/ATMOSPHERICS/components/binary_devices/binary_atmos_base.dm]: qdel

File: code/ATMOSPHERICS/components/binary_devices/binary_atmos_base.dm [hunk 2]
```diff
@@ -12,131 +12,131 @@ obj/machinery/atmospherics/binary
...
+		node1 = null
 
-		else if(reference==node2)
-			del(network2)
-			node2 = null
+	else if(reference==node2)
+		del(network2)
```
Comment (Fox-McCloud) [inline code/ATMOSPHERICS/components/binary_devices/binary_atmos_base.dm]: qdel

File: code/ATMOSPHERICS/components/trinary_devices/trinary_base.dm [hunk 3]
```diff
@@ -17,153 +17,153 @@ obj/machinery/atmospherics/trinary
...
-
-		return results
-
-	disconnect(obj/machinery/atmospherics/reference)
-		if(reference==node1)
-			del(network1)
-			node1 = null
+/obj/machinery/atmospherics/trinary/disconnect(obj/machinery/atmospherics/reference)
+	if(reference==node1)
+		del(network1)
```
Comment (Fox-McCloud) [inline code/ATMOSPHERICS/components/trinary_devices/trinary_base.dm]: qdel

File: code/ATMOSPHERICS/components/trinary_devices/trinary_base.dm [hunk 4]
```diff
@@ -17,153 +17,153 @@ obj/machinery/atmospherics/trinary
...
+		node1 = null
 
-		else if(reference==node2)
-			del(network2)
-			node2 = null
+	else if(reference==node2)
+		del(network2)
```
Comment (Fox-McCloud) [inline code/ATMOSPHERICS/components/trinary_devices/trinary_base.dm]: qdel

File: code/ATMOSPHERICS/components/trinary_devices/trinary_base.dm [hunk 5]
```diff
@@ -17,153 +17,153 @@ obj/machinery/atmospherics/trinary
...
+		node2 = null
 
-		else if(reference==node3)
-			del(network3)
-			node3 = null
+	else if(reference==node3)
+		del(network3)
```
Comment (Fox-McCloud) [inline code/ATMOSPHERICS/components/trinary_devices/trinary_base.dm]: qdel

File: code/ATMOSPHERICS/components/unary/heat_exchanger.dm [hunk 6]
```diff
@@ -10,80 +10,80 @@
...
+		return 1
+	playsound(get_turf(src), 'sound/items/Ratchet.ogg', 50, 1)
+	user << "\blue You begin to unfasten \the [src]..."
+	if (do_after(user, 40))
+		user.visible_message( \
+			"[user] unfastens \the [src].", \
+			"\blue You have unfastened \the [src].", \
+			"You hear ratchet.")
+		new /obj/item/pipe(loc, make_from=src)
+		del(src)
```
Comment (Fox-McCloud) [inline code/ATMOSPHERICS/components/unary/heat_exchanger.dm]: qdel

File: code/ATMOSPHERICS/components/unary/thermal_plate.dm [hunk 7]
```diff
@@ -12,102 +12,101 @@
...
 		return 1
+	playsound(get_turf(src), 'sound/items/Ratchet.ogg', 50, 1)
+	user << "\blue You begin to unfasten \the [src]..."
+	if (do_after(user, 40))
+		user.visible_message( \
+			"[user] unfastens \the [src].", \
+			"\blue You have unfastened \the [src].", \
+			"You hear ratchet.")
+		new /obj/item/pipe(loc, make_from=src)
+		del(src)
```
Comment (Fox-McCloud) [inline code/ATMOSPHERICS/components/unary/thermal_plate.dm]: qdel

File: code/ATMOSPHERICS/pipes/vent.dm [hunk 8]
```diff
@@ -0,0 +1,105 @@
...
+		return 1
+	playsound(T, 'sound/items/Ratchet.ogg', 50, 1)
+	user << "\blue You begin to unfasten \the [src]..."
+	if (do_after(user, 40))
+		user.visible_message( \
+			"[user] unfastens \the [src].", \
+			"\blue You have unfastened \the [src].", \
+			"You hear ratchet.")
+		new /obj/item/pipe(T, make_from=src)
+		del(src)
```
Comment (Fox-McCloud) [inline code/ATMOSPHERICS/pipes/vent.dm]: qdel

File: code/ATMOSPHERICS/pipes/vent.dm [hunk 9]
```diff
@@ -0,0 +1,105 @@
...
+				src.connected_to = c
+				node1 = target
+				break
+
+	update_icon()
+
+/obj/machinery/atmospherics/pipe/vent/disconnect(obj/machinery/atmospherics/reference)
+	if(reference == node1)
+		if(istype(node1, /obj/machinery/atmospherics/pipe))
+			del(parent)
```
Comment (Fox-McCloud) [inline code/ATMOSPHERICS/pipes/vent.dm]: qdel

File: code/ATMOSPHERICS/pipes/tank.dm [hunk 10]
```diff
@@ -0,0 +1,166 @@
...
+				src.connected_to = c
+				node1 = target
+				break
+
+	update_underlays()
+
+/obj/machinery/atmospherics/pipe/tank/disconnect(obj/machinery/atmospherics/reference)
+	if(reference == node1)
+		if(istype(node1, /obj/machinery/atmospherics/pipe))
+			del(parent)
```
Comment (Fox-McCloud) [inline code/ATMOSPHERICS/pipes/tank.dm]: qdel

File: code/ATMOSPHERICS/pipes/simple/pipe_simple_he.dm [hunk 11]
```diff
@@ -0,0 +1,118 @@
...
+		if(target.initialize_directions_he & get_dir(target,src))
+			node1 = target
+			break
+	for(var/obj/machinery/atmospherics/pipe/simple/heat_exchanging/target in get_step(src,node2_dir))
+		if(target.initialize_directions_he & get_dir(target,src))
+			node2 = target
+			break
+
+	if(!node1 && !node2)
+		del(src)
```
Comment (Fox-McCloud) [inline code/ATMOSPHERICS/pipes/simple/pipe_simple_he.dm]: qdel

File: code/ATMOSPHERICS/pipes/simple/pipe_simple.dm [hunk 12]
```diff
@@ -0,0 +1,181 @@
...
+
+/obj/machinery/atmospherics/pipe/simple/disconnect(obj/machinery/atmospherics/reference)
+	if(reference == node1)
+		if(istype(node1, /obj/machinery/atmospherics/pipe))
+			del(parent)
+		node1 = null
+
+	if(reference == node2)
+		if(istype(node2, /obj/machinery/atmospherics/pipe))
+			del(parent)
```
Comment (Fox-McCloud) [inline code/ATMOSPHERICS/pipes/simple/pipe_simple.dm]: qdel

File: code/ATMOSPHERICS/pipes/simple/pipe_simple.dm [hunk 13]
```diff
@@ -0,0 +1,181 @@
...
+
+	var/turf/T = get_turf(src)
+	if(istype(T))
+		hide(T.intact)
+	update_icon()
+
+/obj/machinery/atmospherics/pipe/simple/disconnect(obj/machinery/atmospherics/reference)
+	if(reference == node1)
+		if(istype(node1, /obj/machinery/atmospherics/pipe))
+			del(parent)
```
Comment (Fox-McCloud) [inline code/ATMOSPHERICS/pipes/simple/pipe_simple.dm]: qdel

File: code/ATMOSPHERICS/pipes/simple/pipe_simple.dm [hunk 14]
```diff
@@ -0,0 +1,181 @@
...
+
+	overlays.Cut()
+
+	if(!node1 && !node2)
+		var/turf/T = get_turf(src)
+		new /obj/item/pipe(loc, make_from=src)
+		for (var/obj/machinery/meter/meter in T)
+			if (meter.target == src)
+				new /obj/item/pipe_meter(T)
+				del(meter)
```
Comment (Fox-McCloud) [inline code/ATMOSPHERICS/pipes/simple/pipe_simple.dm]: qdel

File: code/ATMOSPHERICS/pipes/simple/pipe_simple.dm [hunk 15]
```diff
@@ -0,0 +1,181 @@
...
+
+	else return 1
+
+/obj/machinery/atmospherics/pipe/simple/proc/burst()
+	src.visible_message("\red \bold [src] bursts!");
+	playsound(src.loc, 'sound/effects/bang.ogg', 25, 1)
+	var/datum/effect/effect/system/harmless_smoke_spread/smoke = new
+	smoke.set_up(1,0, src.loc, 0)
+	smoke.start()
+	del(src)
```
Comment (Fox-McCloud) [inline code/ATMOSPHERICS/pipes/simple/pipe_simple.dm]: qdel

File: code/ATMOSPHERICS/pipes/pipe.dm [hunk 16]
```diff
@@ -0,0 +1,138 @@
...
+	if (do_after(user, 40))
+		user.visible_message( \
+			"[user] unfastens \the [src].", \
+			"\blue You have unfastened \the [src].", \
+			"You hear ratchet.")
+		var/obj/item/P = new /obj/item/pipe(loc, make_from=src)
+		for (var/obj/machinery/meter/meter in T)
+			if (meter.target == src)
+				new /obj/item/pipe_meter(T)
+				del(meter)
```
Comment (Fox-McCloud) [inline code/ATMOSPHERICS/pipes/pipe.dm]: qdel

File: code/ATMOSPHERICS/pipes/manifold4w.dm [hunk 17]
```diff
@@ -0,0 +1,256 @@
...
+		if(target.initialize_directions & 4)
+			var/c = check_connect_types(target,src)
+			if (c)
+				target.connected_to = c
+				src.connected_to = c
+				node4 = target
+				break
+
+	if(!node1 && !node2 && !node3&& !node4)
+		del(src)
```
Comment (Fox-McCloud) [inline code/ATMOSPHERICS/pipes/manifold4w.dm]: qdel

File: code/ATMOSPHERICS/pipes/manifold4w.dm [hunk 18]
```diff
@@ -0,0 +1,256 @@
...
+
+	alpha = 255
+
+	if(!node1 && !node2 && !node3 && !node4)
+		var/turf/T = get_turf(src)
+		new /obj/item/pipe(loc, make_from=src)
+		for (var/obj/machinery/meter/meter in T)
+			if (meter.target == src)
+				new /obj/item/pipe_meter(T)
+				del(meter)
```
Comment (Fox-McCloud) [inline code/ATMOSPHERICS/pipes/manifold4w.dm]: qdel

File: code/ATMOSPHERICS/pipes/manifold4w.dm [hunk 19]
```diff
@@ -0,0 +1,256 @@
...
+		node2 = null
+
+	if(reference == node3)
+		if(istype(node3, /obj/machinery/atmospherics/pipe))
+			del(parent)
+		node3 = null
+
+	if(reference == node4)
+		if(istype(node4, /obj/machinery/atmospherics/pipe))
+			del(parent)
```
Comment (Fox-McCloud) [inline code/ATMOSPHERICS/pipes/manifold4w.dm]: qdel

File: code/ATMOSPHERICS/pipes/manifold4w.dm [hunk 20]
```diff
@@ -0,0 +1,256 @@
...
+		node1 = null
+
+	if(reference == node2)
+		if(istype(node2, /obj/machinery/atmospherics/pipe))
+			del(parent)
+		node2 = null
+
+	if(reference == node3)
+		if(istype(node3, /obj/machinery/atmospherics/pipe))
+			del(parent)
```
Comment (Fox-McCloud) [inline code/ATMOSPHERICS/pipes/manifold4w.dm]: qdel

File: code/ATMOSPHERICS/pipes/manifold4w.dm [hunk 21]
```diff
@@ -0,0 +1,256 @@
...
+
+/obj/machinery/atmospherics/pipe/manifold4w/disconnect(obj/machinery/atmospherics/reference)
+	if(reference == node1)
+		if(istype(node1, /obj/machinery/atmospherics/pipe))
+			del(parent)
+		node1 = null
+
+	if(reference == node2)
+		if(istype(node2, /obj/machinery/atmospherics/pipe))
+			del(parent)
```
Comment (Fox-McCloud) [inline code/ATMOSPHERICS/pipes/manifold4w.dm]: qdel

File: code/ATMOSPHERICS/pipes/manifold4w.dm [hunk 22]
```diff
@@ -0,0 +1,256 @@
...
+		node3.disconnect(src)
+	if(node4)
+		node4.disconnect(src)
+
+	return ..()
+
+/obj/machinery/atmospherics/pipe/manifold4w/disconnect(obj/machinery/atmospherics/reference)
+	if(reference == node1)
+		if(istype(node1, /obj/machinery/atmospherics/pipe))
+			del(parent)
```
Comment (Fox-McCloud) [inline code/ATMOSPHERICS/pipes/manifold4w.dm]: qdel

File: code/ATMOSPHERICS/pipes/manifold.dm [hunk 23]
```diff
@@ -0,0 +1,251 @@
...
+						target.connected_to = c
+						src.connected_to = c
+						node3 = target
+						connect_directions &= ~direction
+						break
+			if (node3)
+				break
+
+	if(!node1 && !node2 && !node3)
+		del(src)
```
Comment (Fox-McCloud) [inline code/ATMOSPHERICS/pipes/manifold.dm]: qdel

File: code/ATMOSPHERICS/components/unary/unary_base.dm [hunk 24]
```diff
@@ -9,88 +9,88 @@
...
-		return results
+	return results
 
-	disconnect(obj/machinery/atmospherics/reference)
-		if(reference==node)
-			del(network)
-			node = null
+/obj/machinery/atmospherics/unary/disconnect(obj/machinery/atmospherics/reference)
+	if(reference==node)
+		del(network)
```
Comment (Fox-McCloud) [inline code/ATMOSPHERICS/components/unary/unary_base.dm]: qdel

File: code/ATMOSPHERICS/pipes/cap.dm [hunk 25]
```diff
@@ -0,0 +1,122 @@
...
+/obj/machinery/atmospherics/pipe/cap/Destroy()
+	if(node)
+		node.disconnect(src)
+
+	return ..()
+
+/obj/machinery/atmospherics/pipe/cap/disconnect(obj/machinery/atmospherics/reference)
+	if(reference == node)
+		if(istype(node, /obj/machinery/atmospherics/pipe))
+			del(parent)
```
Comment (Fox-McCloud) [inline code/ATMOSPHERICS/pipes/cap.dm]: qdel

File: code/ATMOSPHERICS/pipes/manifold.dm [hunk 26]
```diff
@@ -0,0 +1,251 @@
...
+		node2.disconnect(src)
+	if(node3)
+		node3.disconnect(src)
+
+	return ..()
+
+/obj/machinery/atmospherics/pipe/manifold/disconnect(obj/machinery/atmospherics/reference)
+	if(reference == node1)
+		if(istype(node1, /obj/machinery/atmospherics/pipe))
+			del(parent)
```
Comment (Fox-McCloud) [inline code/ATMOSPHERICS/pipes/manifold.dm]: qdel

File: code/ATMOSPHERICS/pipes/manifold.dm [hunk 27]
```diff
@@ -0,0 +1,251 @@
...
+		node1 = null
+
+	if(reference == node2)
+		if(istype(node2, /obj/machinery/atmospherics/pipe))
+			del(parent)
```
Comment (Fox-McCloud) [inline code/ATMOSPHERICS/pipes/manifold.dm]: qdel

File: code/ATMOSPHERICS/pipes/manifold.dm [hunk 28]
```diff
@@ -0,0 +1,251 @@
...
+		node2 = null
+
+	if(reference == node3)
+		if(istype(node3, /obj/machinery/atmospherics/pipe))
+			del(parent)
```
Comment (Fox-McCloud) [inline code/ATMOSPHERICS/pipes/manifold.dm]: qdel

File: code/ATMOSPHERICS/pipes/manifold.dm [hunk 29]
```diff
@@ -0,0 +1,251 @@
...
+
+	alpha = 255
+
+	if(!node1 && !node2 && !node3)
+		var/turf/T = get_turf(src)
+		new /obj/item/pipe(loc, make_from=src)
+		for (var/obj/machinery/meter/meter in T)
+			if (meter.target == src)
+				new /obj/item/pipe_meter(T)
+				del(meter)
```
Comment (Fox-McCloud) [inline code/ATMOSPHERICS/pipes/manifold.dm]: qdel

**Comment by Fox-McCloud** [comment]
there probably needs to be some gcdestroyed in a few places, but @Krausus would be the better one for that.

**Comment by Fox-McCloud** [comment]
:+1: 

I didn't notice anything.

**Comment by PsiOmegaDelta** [comment]
If this is as good as I hope it is then it deserves a port.

**Comment by Fox-McCloud** [comment]
@PsiOmegaDelta 

Very little was actually changed, aside from using qdel over del, correcting some grammar issues, and updating to span classes.

Overall, the code is functionally the same, it just looks a lot better.

**Pull request merged**